core/src/main/java/org/bitcoinj/core/Context.java

```diff
@@ -377,7 +377,11 @@ public void close() {
             if(masternodeSync.hasSyncFlag(MasternodeSync.SYNC_FLAGS.SYNC_INSTANTSENDLOCKS))
                 llmqBackgroundThread.interrupt();
             blockChain.removeNewBestBlockListener(newBestBlockListener);
-            scheduledMasternodeSync.cancel(true);
+            if (scheduledMasternodeSync != null)
+                scheduledMasternodeSync.cancel(true);
+            blockChain.close();
+            if (headerChain != null)
+                headerChain.close();
```

Comment:
This function is now called by Dash Wallet (another PR) when it disconnects from the network and this will avoid memory leaks where an AbstractBlockChain object remains alive with a closed BlockStore.